### ISSUE TITLE
Pagefind UI: Focus the search input

### ISSUE THREAD

**Issue by bglw** [CloudCannon/pagefind#121]
### Discussed in https://github.com/CloudCannon/pagefind/discussions/120

<div type='discussions-op-text'>

<sup>Originally posted by **dprothero** October 27, 2022</sup>
I'm putting the Pagefind UI within a modal dialog that opens up when someone clicks a Search button in the navbar of my site. Right now, I have to do some hacky stuff to wait for the search input to be rendered and then add focus to it.

Would it make sense to have an option to automatically focus the search input? I'm not super familiar with Svelte, but it seems it would be easier to do it within the Pagefind UI. On the other hand, I don't want to pollute the API with too many options.</div>

<hr/>

**Feature:** Implement an `autoFocus` option for Pagefind UI, as well as a `focus()` function on the UI object.

**Comment by ndeville** [comment]
+1 from me. 

My use case: 
static website acting as my online notebook. 
I use my own site everyday a lot to access/search my notes. 
With previous search engine, I had the cursor/focus set on the search field when opening the page. 
So my workflow is a keyboard shortcut to open my site and being able to immediately search a note - instead of clicking in the field or using tab to get there.

I would add a second nice-to-have feature which is keyboard navigation of search results.
So once I hit enter on my search, I can tab my way through result list and hit enter to open the result I wish.

**Comment by bglw** [comment]
> I would add a second nice-to-have feature which is keyboard navigation of search results

I don't think this is something I'll tackle in the official Pagefind UI — there would need to be accessibility work done around this since normal anchor elements aren't intended to be part of the tab order — I think at this level Pagefind will defer to people building their own search components on top of the JS API. 

Actions like focusing the input get the green light, though 👍 I'll make sure to get to this soon.

**Comment by ndeville** [comment]
I will look into how the keyboard/tab navigation was implemented with Pelican - see if I can replicate it myself with Pagefind.

Separate question: would passing a search query via URL parameter be possible?

**Comment by bglw** [comment]
@ndeville There's some discussion of this in #144 — it is possible but you'll need a small code snippet of your own to get the query from the URL 🙂
```js
let pagefind = new PagefindUI({ element: "#search" });
pagefind.triggerSearch("some search terms");
```

**Comment by davidsneighbour** [comment]
There is an HTML attribute `autofocus` that should do the trick? Why not add it at least to the PagefindUI as this seems to be something that is designed to be the focal point of a page that it is one (as opposed to a search field somewhere hidden in the navigation bar)?

**Comment by vanbroup** [comment]
@bglw can you make this a configurable option? The inability to autofocus creates a common accessibility issue, for example when the search is loaded into a modal view.

I agree that you might not want to make this the default behavior as it could negatively impact the accessibility of the site depending on where it's loaded.

**Comment by vanbroup** [comment]
We have the same accessibility issue (lacking autofocus) when opening the search into a dialog element, it would be great of this could be addressed. 

@bglw any chance to get this implemented soon?

A configurable [autofocus](https://developer.mozilla.org/en-US/docs/Web/HTML/Global_attributes/autofocus) as suggested by @davidsneighbour seems like a simple approach.

**Comment by bglw** [comment]
Hi all 👋 

Thanks to @vanbroup this is now available in the `1.0.5-rc0` release 🙂

**Comment by bglw** [comment]
This has now landed in the [v1.1.0 release](https://github.com/CloudCannon/pagefind/releases/tag/v1.1.0) 🎉
